torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+                        out_dtype="int32",
+                        resampling=Resampling.bilinear,
+                    )
+                image = torch.from_numpy(array).float()
+                c = img_idx * self.channels + bnd_idx
+                tensor[c : c + image.shape[0]] = image
+                bnd_idx += image.shape[0]
+        return tensor
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load the target mask for a single location.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask
+        """
+        path = self.files.iloc[index][0]
+        with rasterio.open(f"{path}_labels.tif") as f:
+            array = f.read() - 1
+        tensor = torch.from_numpy(array).squeeze().long()
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if all files already exist
+        if all(
+            os.path.exists(os.path.join(self.root, name))
+            for name in self.extracted_names
+        ):
+            return
+
+        # Check for downloaded files and extract existing zips
+        missing = []
+        for file_name in self.file_names:
+            file_path = os.path.join(self.root, file_name)
+            if not os.path.exists(file_path):
+                missing.append(file_name)
+            elif file_name[-3:] == "zip":
+                extract_archive(file_path)
+
+        if not missing:
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            missing_str = (
+                f"{', '.join(missing[:-1])} and {missing[-1]}"
+                if len(missing) > 1
+                else missing[0]
+            )
+            raise RuntimeError(
+                f"{missing_str} not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True`"
+                " to automatically download the dataset."
+            )
+
+        # Download missing files
+        record_info = requests.get(self.url).json()["files"]
+        extractable = []
+        for file_info in record_info:
+            file_path = os.path.join(self.root, file_info["key"])
+            if not os.path.exists(file_path):
+                download_url(
+                    file_info["links"]["self"],
+                    self.root,
+                    filename=file_info["key"],
+                    md5=file_info["checksum"].split(":")[1] if self.checksum else None,
+                )
+                if file_info["type"] == "zip":
+                    extractable.append(file_path)
+
+        # Extract downloaded files
+        for file_path in extractable:
+            extract_archive(file_path)
+
+    def plot(
```

Comment:
Can you add an example plot to the PR description so we can check if plotting works correctly?

Reply:
Done